src/views/ScaleManagerDemo.vue

```diff
@@ -0,0 +1,295 @@
+          'value': 660,
+          'label': 'Safe Height',
+          'rules':         [
+            v => (v && v > 0) || 'Safe Height must be greater than 0',
+            v => (v < this.scaleValues.maxHeight.value) || 'Safe height must be less than max height'
+          ],
+          'hint': 'The minimum vertical dimension in pixel of the safe area in your game.'
+        },
+      },
+      anchorValues: {
+        'position': {
+          'value': {'x': 0, 'y': 0},
+          'label': 'Anchor Position'
+        },
+        'direction': {
+          'value': {'x': 0, 'y': 0},
+          'label': 'Anchor Direction'
+        },
+      },
+      isFullScreen: false,
+      anchorDirectionRules: [
+        v => (v <= 1 && v >= -1) || 'Anchor Directions must be between -1 and 1'
+      ],
+      scalingRules: [
+        [
+          v => (v && v > 0) || 'Max Width must be greater than 0',
+          v => (v > this.scaleValues.safeWidth.value) || 'Max width must be more than safe width'
+        ],
+        [
+          v => (v && v > 0) || 'Max Height must be greater than 0',
+          v => (v > this.scaleValues.safeHeight.value) || 'Max height must be more than safe height'
+        ],
+        [
+          v => (v && v > 0) || 'Safe Width must be greater than 0',
+          v => (v < this.scaleValues.maxWidth.value) || 'Safe width must be less than max width'
+        ],
+        [
+          v => (v && v > 0) || 'Safe Height must be greater than 0',
+          v => (v < this.scaleValues.maxHeight.value) || 'Safe height must be less than max height'
+        ],
+      ],
+    };
+  },
+  methods: {
+    fullScreen() {
+      const elem = document.querySelector('iframe');
+      if (!document.fullscreenElement) {
+        elem.requestFullscreen().catch(err => {
+          alert(`Error attempting to enable full-screen mode: ${err.message} (${err.name})`);
+        });
+      } else {
+        document.exitFullscreen();
+      }
+    },
+    updateScaling() {
+      this.bellhop.send('applyChanges', {
+        'maxWidth': +this.scaleValues.maxWidth.value || 1320,
+        'maxHeight': +this.scaleValues.maxHeight.value || 780,
+        'safeWidth': +this.scaleValues.safeWidth.value || 1024,
+        'safeHeight': +this.scaleValues.safeHeight.value || 660,
+        'position': { 'x': +this.anchorValues.position.value.x, 'y': +this.anchorValues.position.value.y },
+        'direction': { 'x': +this.anchorValues.direction.value.x, 'y': +this.anchorValues.direction.value.y },
+      });
+    },
+    updateAnchor() {
+      this.bellhop.send('updateAnchor', {
+        'position': { 'x': +this.anchorValues.position.value.x, 'y': +this.anchorValues.position.value.y },
+        'direction': { 'x': +this.anchorValues.direction.value.x, 'y': +this.anchorValues.direction.value.y }
+      });
+    },
+    validateAnchor () {
+      if (!this.$refs.anchorForm.validate()) {
+        return;
+      }
+
+      this.updateAnchor();
+    },
+    validateScale () {
```

Comment:
Do we actually need to tie this to a button click? Can we just run this on `change` events from each of the elements?

Reply:
Updated to have the anchor update in real time. The scaling values (width/height etc) should probably be tied to the update button as the game has to re-initialize every time they change. But I'm unsure of how to convey the difference in behavior.